client/app/src/main/java/com/microsoft/research/karya/ui/scenarios/common/MicrotaskRenderer.kt

```diff
@@ -319,14 +326,11 @@ abstract class MicrotaskRenderer(
       /** Fetch the task */
       ioScope.launch { task = karyaDb.taskDao().getById(taskId) }.join()
 
-      // TODO: Maybe pass an intent extra rather than storing in database
       ioScope.launch {
-        firstTimeActivityVisit =
-          try {
-            !thisWorker.params!!.asJsonObject.get(activityName).asBoolean
-          } catch (e: Exception) {
-            true
-          }
+        val firstRunKey = booleanPreferencesKey(PreferenceKeys.IS_FIRST_RUN)
+        val data = applicationContext.dataStore.data.first()
+        firstTimeActivityVisit = data[firstRunKey] ?: true
+        applicationContext.dataStore.edit { prefs -> prefs[firstRunKey] = false }
```

Comment:
Hey, can you change this? The first run prompt is useful for temp data storage and we're gonna mark it as false after that. You can create a new `PreferenceKey` for something like `SPEECH_ACTIVITY_VISITED`.